test/curve_analysis/test_curve_fit.py

```diff
@@ -571,3 +554,182 @@ def test_fixed_param_is_missing(self):
 
         with self.assertRaises(AnalysisError):
             analysis._run_analysis(test_data, **default_opts.__dict__)
+
+
+class TestFitOptions(QiskitTestCase):
+    """Unittest for fit option object."""
+
+    def test_empty(self):
+        """Test if default value is automatically filled."""
+        opt = FitOptions(["p0", "p1", "p2"])
+
+        # bounds should be default to inf tuple. otherwise crashes the scipy fitter.
+        ref_opts = {
+            "p0": {"p0": None, "p1": None, "p2": None},
+            "bounds": {"p0": (-np.inf, np.inf), "p1": (-np.inf, np.inf), "p2": (-np.inf, np.inf)},
+        }
+
+        self.assertDictEqual(opt.options, ref_opts)
+
+    def test_create_option_with_dict(self):
+        """Create option and fill with dictionary."""
+        opt = FitOptions(
+            ["p0", "p1", "p2"],
+            default_p0={"p0": 0, "p1": 1, "p2": 2},
+            default_bounds={"p0": (0, 1), "p1": (1, 2), "p2": (2, 3)},
+        )
+
+        ref_opts = {
+            "p0": {"p0": 0.0, "p1": 1.0, "p2": 2.0},
+            "bounds": {"p0": (0.0, 1.0), "p1": (1.0, 2.0), "p2": (2.0, 3.0)},
+        }
+
+        self.assertDictEqual(opt.options, ref_opts)
+
+    def test_create_option_with_array(self):
+        """Create option and fill with array."""
+        opt = FitOptions(
+            ["p0", "p1", "p2"],
+            default_p0=[0, 1, 2],
+            default_bounds=[(0, 1), (1, 2), (2, 3)],
+        )
+
+        ref_opts = {
+            "p0": {"p0": 0.0, "p1": 1.0, "p2": 2.0},
+            "bounds": {"p0": (0.0, 1.0), "p1": (1.0, 2.0), "p2": (2.0, 3.0)},
+        }
+
+        self.assertDictEqual(opt.options, ref_opts)
+
+    def test_override_partial_dict(self):
+        """Create option and override value with partial dictionary."""
+        opt = FitOptions(["p0", "p1", "p2"])
+        opt.p0["p1"] = 3
+
+        ref_opts = {
+            "p0": {"p0": None, "p1": 3, "p2": None},
+            "bounds": {"p0": (-np.inf, np.inf), "p1": (-np.inf, np.inf), "p2": (-np.inf, np.inf)},
+        }
+
+        self.assertDictEqual(opt.options, ref_opts)
+
+    def test_cannot_override_assigned_value(self):
+        """Test cannot override already assigned value."""
+        opt = FitOptions(["p0", "p1", "p2"])
+        opt.p0["p1"] = 3
+        opt.p0["p1"] = 5
+
+        ref_opts = {
+            "p0": {"p0": None, "p1": 3, "p2": None},
+            "bounds": {"p0": (-np.inf, np.inf), "p1": (-np.inf, np.inf), "p2": (-np.inf, np.inf)},
+        }
```

Comment:
This looks like the part that Yael was referring to. It does seem strange to assign 5 to "p1" and see that nothing happens.

Reply:
Explicit method is added https://github.com/Qiskit/qiskit-experiments/pull/390/commits/02279b7d91bc449559fc6a07b506917a80236c92

Developers need to remember to use this method to protect user guesses.
https://github.com/Qiskit/qiskit-experiments/blob/b039555f46e37c69d92a29186a9c0d58bb06d2d7/test/curve_analysis/test_curve_fit.py#L723